### PR TITLE
Verify google code format

### PR DESCRIPTION
**Pull request by oliemansm**
Added plugin to verify google code format and updated the Contributions file explaining how to fix code formatting if the build fails because of the check.

It's as simple as running:
```bash
./gradlew googleJavaFormat
```

### CODE REVIEW

**Comment by setchy** [comment]
I'm struggling to remember the exact name of the tool, but I've seen other OSS repos where they commit their coding standards and IDE formatters to the root of their project.  Could this be also useful in addition to the bot

**Comment by oliemansm** [comment]
@setchy Looks like Spring Boot also has folders in the root for `idea` and `eclipse` with their code style settings: https://github.com/spring-projects/spring-boot. The contributions file doesn't really specify how to use those.

I've seen other projects where the default .idea folder with just the code style settings is part of the repo and all other files are ignored. That should automatically apply those code style settings to your project without needing to specifically import them. I could give something like that a try?

**Comment by oliemansm** [comment]
The github action nicely fails showing clearly what's wrong, so I'll go ahead and fix the formatting in this one now:

image

**Comment by oliemansm** [comment]
@setchy I've included the `.idea/codeStyles` folder now which contains the project code style settings in line with the bot. This should result in people using IntelliJ to automatically have the correct code style settings. 

A lot of changed files now because of simple formatting changes.

File: example-spring-common/src/main/java/graphql/kickstart/spring/web/boot/sample/ApplicationBootConfiguration.java [hunk 1]
```diff
@@ -22,9 +22,7 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
-/**
- * @author <a href="mailto:java.lang.RuntimeException@gmail.com">oEmbedler Inc.</a>
- */
+/** @author <a href="mailto:java.lang.RuntimeException@gmail.com">oEmbedler Inc.</a> */
```
Comment (BlasiusSecundus) [inline example-spring-common/src/main/java/graphql/kickstart/spring/web/boot/sample/ApplicationBootConfiguration.java:25]: Nice email address :)

File: graphql-kickstart-spring-boot-autoconfigure-graphql-annotations/src/main/java/graphql/kickstart/graphql/annotations/GraphQLAnnotationsAutoConfiguration.java [hunk 2]
```diff
@@ -179,30 +195,32 @@ private void setQueryResolverClass(
    * This is required, because normally implementations of interfaces are not explicitly returned by
    * any resolver method, and therefor not added to the schema automatically.
    *
-   * All interfaces are considered GraphQL interfaces if they are declared in the configured package
-   * and have at least one {@link GraphQLField}-annotated methods.
+   * <p>All interfaces are considered GraphQL interfaces if they are declared in the configured
```
Comment (BlasiusSecundus) [inline graphql-kickstart-spring-boot-autoconfigure-graphql-annotations/src/main/java/graphql/kickstart/graphql/annotations/GraphQLAnnotationsAutoConfiguration.java:198]: Is the closing tag missing intentionally?

Comment (oliemansm) [reply graphql-kickstart-spring-boot-autoconfigure-graphql-annotations/src/main/java/graphql/kickstart/graphql/annotations/GraphQLAnnotationsAutoConfiguration.java:198]: The google-java-format plugin made this change automatically actually when I ran `./gradlew googleJavaFormat`. So it seems to have been coded this way intentionally by Google.

File: example-spring-common/src/main/java/graphql/kickstart/spring/web/boot/sample/ApplicationBootConfiguration.java (hunk 1 above)
Comment (setchy) [reply example-spring-common/src/main/java/graphql/kickstart/spring/web/boot/sample/ApplicationBootConfiguration.java:25]: 🤣

**Comment by oliemansm** [comment]
Looks like I'm not able to get IntelliJ to play completely nice with that gradle plug-in. Which is very annoying, because now I have to run that gradle task every time to correct it instead of being able to rely on IntelliJ reformatting it the same way. Not sure if this is really the way to go then... 🤔

**Comment by setchy** [comment]
Damn!

I haven't used this myself, but found it yesterday - https://github.com/jhipster/prettier-java - might be worth a look

**Comment by oliemansm** [comment]
Thanks, I'll check it out. I'll also try to see if there's one common configuration to achieve using checkstyle.

**Comment by setchy** [comment]
https://github.com/google/google-java-format?

**Comment by oliemansm** [comment]
Yeah that could work! The gradle plugin is already what this PR is using now. So that would mean that we'd have to add a section in the contributions doc explaining to install that plugin for your IDE too to make it all work. Haven't checked it out how that plugin works in the IDE.

**Comment by setchy** [comment]
I just tested it in IntelliJ.  

Installed plugin, restart IDE, enable it per project (very nice) then Command+Option+L to reformat.  It mentions the plugin doesn't handle organizing imports.  To do that you need to set the File->Settings->Editor->Code Style.  This might be where we use a Project specific file standard that we commit to git

**Comment by setchy** [comment]
I noticed the contributing.md mentions spectrum. I can update the wording tomorrow if no one beats me to it

**Pull request merged**